src/store/store.ts

```diff
@@ -18,10 +20,12 @@ const rootReducer = combineReducers({
 const persistedReducer = persistReducer(persistConfig, rootReducer);
 
 export const store = configureStore({
-  reducer: persistedReducer,
+  reducer: { persistedReducer, [pokemonApi.reducerPath]: pokemonApi.reducer },
+  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(pokemonApi.middleware),
```

Comment:
please move pokemonApi.middleware to constant (array), because we'll have more middlewares later, no point to concat each of them 1 by 1